spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/handler/support/RoutePredicateFactoryUtils.java

```diff
@@ -17,14 +17,25 @@
 
 package org.springframework.cloud.gateway.handler.support;
 
+import java.net.InetSocketAddress;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.apache.logging.log4j.util.Strings;
 import org.springframework.cloud.gateway.handler.predicate.RoutePredicateFactory;
+import org.springframework.web.server.ServerWebExchange;
 
 /**
  * @author Spencer Gibb
+ * @author Andrew Fitzgerald
  */
 public class RoutePredicateFactoryUtils {
```

Comment:
I don't think this is the right place for this. After looking at the class, I'm going to remove it and move the one static method to the single class where it is used.

Reply:
Maybe a class called `RemoteAddressResolver`.